### PR TITLE
Refs #30288 - improves CPUs counter input

### PR DESCRIPTION
**Pull request by ezr-ondrej**
In 22fa163731b38f597e4f96e0c0b345518a360b37 we've introduced React
component to implement the counter funcionality.
This uses the old `counter_f` form helper to render the React component
and effectivelly remove our dependency on the ui.spinner library.

### CODE REVIEW

**Review by yifatmakias — DISMISSED** [review]
Thanks, @ezr-ondrej. It works for me, checked in plugin as well.

**Comment by ezr-ondrej** [comment]
> Thanks, @ezr-ondrej. It works for me, checked in plugin as well.

Awesome, thanks @yifatmakias :tada:
I've rebased, so the tests should pass now.

**Comment by ezr-ondrej** [comment]
[test katello] got stuck

**Comment by ezr-ondrej** [comment]
:green_apple:

File: webpack/assets/javascripts/react_app/components/common/forms/InputFactory.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import DateTimePicker from '../DateTimePicker/DateTimePicker';
 import DatePicker from '../DateTimePicker/DatePicker';
 import OrderableSelect from './OrderableSelect';
 import MemoryAllocationInput from '../../MemoryAllocationInput/MemoryAllocationInput';
+import CounterInput from './CounterInput/CounterInput';
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/common/forms/InputFactory.js]: there's an index file, so 
```js
import CounterInput from './CounterInput';
```
should be enough

File: webpack/assets/javascripts/react_app/components/common/forms/CounterInput/CounterInput.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      setWarning(null);
+      setError(__('Specified value is higher than maximum value'));
+    } else if (recommendedMaxValue && innerValue > recommendedMaxValue) {
+      setError(null);
+      setWarning(__('Specified value is higher than recommended maximum'));
+    } else {
+      setError(null);
+      setWarning(null);
+    }
+  }, [recommendedMaxValue, maxValue, innerValue, setError, setWarning]);
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/common/forms/CounterInput/CounterInput.js]: nit: we can ommit the `setState` (setError, setWarning) setters from the dependency list, react guarantees that `setState` is stable during renders.

https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies
https://atomizedobjects.com/blog/react/how-to-use-setstate-from-within-useeffect-with-dependencies/

File: webpack/assets/javascripts/react_app/components/common/forms/CounterInput/CounterInput.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+  return (
+    <RCInputNumber
+      value={innerValue}
+      name={name}
+      id={id}
+      min={minValue}
+      disabled={disabled}
+      onChange={handleChange}
+      step={1}
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/common/forms/CounterInput/CounterInput.js]: the `step` here is constant, though in some usages the step is dynamic, i.e - https://github.com/theforeman/foreman_discovery/blob/19f3cdb8cd171749d98150e2d2f4a7304e93ec59/app/views/discovery_rules/_form.html.erb#L27

File: app/helpers/form_helper.rb [hunk 4]
```diff
@@ -236,18 +236,7 @@ def byte_size_f(f, attr, options = {})
...
-    text_f(f, attr, options)
-  end
-
-  def soft_limit_warning_block
-    content_tag(:span, :class => 'maximum-limit hidden') do
-      icon_text('warning-triangle-o',
-        content_tag(:span, ' ' + _('Specified value is higher than recommended maximum'), :class => 'error-message'),
-        :kind => 'pficon')
-    end
+    react_form_input('counter', f, attr, options)
```
Comment (amirfefer) [inline app/helpers/form_helper.rb:239]: we use `counter_f` in other plugins:
https://github.com/search?q=org%3Atheforeman++counter_f&type=code
is there any regression expected from this? (I've found a regression regarding the `step` prop. as you can see down below)
would it be safer to add a deprecation warning and create a `counter_f_react` for now?

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @ezr-ondrej  👍 
I left a few inline comments

File: webpack/assets/javascripts/react_app/components/common/forms/CounterInput/CounterInput.js (hunk 3 above)
Comment (ezr-ondrej) [reply webpack/assets/javascripts/react_app/components/common/forms/CounterInput/CounterInput.js]: Nice catch, I didn't realize that :+1:

File: app/helpers/form_helper.rb (hunk 4 above)
Comment (ezr-ondrej) [reply app/helpers/form_helper.rb:239]: I don't believe so, we should make sure we keep the same API, but keeping the same helper is very benefitial for the plugins - they don't need to change the usage.
Regressions are always possible, though we should try our best and if plugins find some, they should fix it in here.
If we create another helper, we would force them to switch to that helper and then switch back to the previous name.
I believe the regressions are simple to solve, thus we should be fine fixing them as they come IMHO.

**Comment by tbrisker** [comment]
rebased to fix the merge conflicts

**Review by amirfefer — APPROVED** [review]
Tested with some plugins, works as expected
Thanks @ezr-ondrej !

**Pull request merged**